skymp5-server/cpp/lib_espm/impl/MappedBuffer.cpp

```diff
@@ -0,0 +1,99 @@
+#include "MappedBuffer.h"
+
+#include <fmt/format.h>
+
+#ifndef WIN32
+#  include <iostream>
+
+// Linux
+#  include <fcntl.h>
+#  include <sys/mman.h>
+#  include <unistd.h>
+#endif
+
+namespace espm::impl {
+
+MappedBuffer::MappedBuffer(const fs::path& path)
+{
+  size_ = fs::file_size(path);
+#ifdef WIN32
+  fileHandle_ = CreateFileW(path.c_str(), GENERIC_READ, FILE_SHARE_READ, NULL,
+                            OPEN_EXISTING, FILE_FLAG_RANDOM_ACCESS, 0);
+  if (!fileHandle_) {
+    // generic vs system?
+    throw std::system_error(
+      std::error_code(GetLastError(), std::system_category()),
+      "CreateFileW failed");
+  }
+
+  mapHandle_ = CreateFileMapping(fileHandle_, NULL, PAGE_READONLY, 0, 0, NULL);
+  if (!mapHandle_) {
+    throw std::system_error(
+      std::error_code(GetLastError(), std::system_category()),
+      "CreateFileMapping failed");
+  }
+
+  viewPtr_ = MapViewOfFile(mapHandle_, FILE_MAP_READ, 0, 0, 0);
+  if (!mapHandle_) {
+    throw std::system_error(
+      std::error_code(GetLastError(), std::system_category()),
+      "CreateFileMapping failed");
+  }
+  data_ = static_cast<char*>(viewPtr_);
+#else
+  fd_ = open(path.c_str(), O_RDONLY);
+  if (fd_ == -1) {
+    throw std::system_error(errno, std::generic_category(),
+                            fmt::format("Can't read {}", path.string()));
+  }
+  const auto mmapResult = mmap(NULL, size_, PROT_READ, MAP_SHARED, fd_, 0);
+  if (mmapResult == MAP_FAILED) {
+    throw std::system_error(errno, std::generic_category(),
+                            fmt::format("Can't map {}", path.string()));
+  }
+  data_ = static_cast<char*>(mmapResult);
+#endif
+}
+
+MappedBuffer::~MappedBuffer()
```

Comment:
I am not sure this destructor would be called in unit tests so consider checking.

Reply:
It should, as part of `espm::Loader`'s destructor (and then, destructor of this field: `std::unique_ptr<impl::IBuffer> buffer;`), but ok, I'll double-check just in case

Reply:
Yep, this is called. Changed `unmap`'s return value to -1, and program was aborted.